ci/deploy.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -e
+#
+# Deploy your branch on VIP Go.
+#
+
+# This script uses various Circle CI and Travis CI environment 
+# variables, Circle prefix their environment variables with 
+# `CIRCLE_` and Travis with `TRAVIS_`. 
+# Documentation:
+# https://circleci.com/docs/1.0/environment-variables/
+# https://docs.travis-ci.com/user/environment-variables/
+
+set -ex
+
+DEPLOY_SUFFIX="-built"
+
+BRANCH="${CIRCLE_BRANCH:-$TRAVIS_BRANCH}"
+SRC_DIR="${TRAVIS_BUILD_DIR:-$PWD}"
+BUILD_DIR="/tmp/vip-go-build"
+
+if [[ -z "$BRANCH" ]]; then
+	echo "No branch specified!"
+	exit 1
+fi
+
+if [[ -d "$BUILD_DIR" ]]; then
+	echo "WARNING: ${BUILD_DIR} already exists. You may have accidentally cached this"
```

Comment:
Should we prefix the BUILD_DIR with a timestamp to minimize potential failures due to caching?

Reply:
I'm not sure about this, if we do this and the directory does get cached then we'll gradually fill the cache will directories with timestamps in the name. What do you think, @mjangda ?

Reply:
I think the space issue is more a problem for the CI service (or the server running the script) to handle. We should optimize for reliable deploys (making sure we push out the right thing and try to limit failures, which can be annoying to recover from).

Reply:
Sold. I've added a timestamp.